### PR TITLE
chore(debugging): ensure correct log probe configuration handling

### PR DESCRIPTION
**Pull request by P403n1x87**
This change ensures that the RCM adapter for dynamic instrumentation can parse the payload attributes correctly.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: tests/debugging/probe/test_remoteconfig.py [hunk 1]
```diff
@@ -260,3 +265,34 @@ def validate_events(expected):
...
+
+
+def test_log_probe_attributes_parsing():
+    probe = probe_factory(
+        {
+            "id": "3d338829-21c4-4a8a-8a1a-71fbce995efa",
+            "version": 0,
+            "type": ProbeType.LOG_PROBE,
+            "language": "python",
+            "active": True,
```
Comment (shatzi) [inline tests/debugging/probe/test_remoteconfig.py:277]: nit: remove active once the #5142 is merged

**Review by shatzi — DISMISSED** [review]
Looks good. nice refactoring

**Review by emmettbutler — APPROVED** [review]

File: ddtrace/debugging/_probe/remoteconfig.py [hunk 2]
```diff
@@ -116,20 +121,38 @@ def _create_probe_based_on_location(args, attribs, line_class, function_class):
...
+        _id = attribs["id"]
+    except KeyError as e:
+        raise ValueError("Invalid probe attributes: %s" % e)
+
+    if _type == ProbeType.LOG_PROBE:
         take_snapshot = attribs.get("captureSnapshot", False)
 
         args = dict(
-            probe_id=_id,
+            probe_id=attribs["id"],
```
Comment (P403n1x87) [inline ddtrace/debugging/_probe/remoteconfig.py:139]:
```suggestion
            probe_id=_id,
```

**Review by shatzi — DISMISSED** [review]

File: tests/debugging/probe/test_remoteconfig.py (hunk 1 above)
Comment (shatzi) [reply tests/debugging/probe/test_remoteconfig.py:272]: merged

**Review by shatzi — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**